examples/audio/ctc_asr.py

```diff
@@ -0,0 +1,526 @@
+
+We will evaluate the quality of the model using [Word Error Rate
+(WER)](https://en.wikipedia.org/wiki/Word_error_rate). WER is obtained by adding up
+the substitutions, insertions, and deletions that occur in a sequence of
+recognized words. Divide that number by the total number of words originally
+spoken. The result is the WER. To get the WER score you need to install the
+[jiwer](https://pypi.org/project/jiwer/) package. Use the following command line:
+
+```
+> pip install jiwer
+
+```
+
+
+
+
+
+**References:**
+
+- [LJSpeech Dataset](https://keithito.com/LJ-Speech-Dataset/)
+- [Speech recognition](https://en.wikipedia.org/wiki/Speech_recognition)
+- [Sequence Modeling With CTC](https://distill.pub/2017/ctc/)
+- [DeepSpeech2](https://nvidia.github.io/OpenSeq2Seq/html/speech-recognition/deepspeech2.html)
+
+"""
+
+"""
+## Setup
+"""
+
+import pandas as pd
+import numpy as np
+from tensorflow.keras import layers
+from tensorflow import keras
+import tensorflow as tf
+from scipy.io import wavfile
+import python_speech_features
+from jiwer import wer
+
+
+"""
+## Load the LJSpeech Dataset
+
+Let's download the [LJSpeech Dataset](https://keithito.com/LJ-Speech-Dataset/).
+The dataset contains 13,100 audio files as `wav` files in the `/wavs/`.
+The label (transcript) for each audio file is a string
+given in the `metadata.csv` file. The fields are:
+
+- `ID:` this is the name of the corresponding .wav file
+- `Transcription:` words spoken by the reader (UTF-8)
+- `Normalized Transcription:` transcription with numbers,
+ordinals, and monetary units expanded into full words b(UTF-8).
+
+For this demo we will use on the `Normalized Transcription` field.
+
+Each audio file is a single-channel 16-bit PCM WAV with a sample rate of 22050 Hz.
+
+
+"""
+
+data_url = "https://data.keithito.com/data/speech/LJSpeech-1.1.tar.bz2"
+data_path = keras.utils.get_file("LJSpeech-1.1", data_url, untar=True)
+wavs_path = data_path + "/wavs/"
+metadata_path = data_path + "/metadata.csv"
+
+
+# Read metadata file and parse it
+metadata_df = pd.read_csv(metadata_path, sep="|", header=None, quoting=3)
+metadata_df.columns = ["file_name", "transcription", "normalized_transcription"]
+metadata_df = metadata_df[["file_name", "normalized_transcription"]]
+metadata_df = metadata_df.sample(frac=1).reset_index(drop=True)
+metadata_df.head(3)
+
+
+"""
+We now split the data into training and validation set.
+
+"""
+
+split = int(len(metadata_df) * 0.90)
+df_train = metadata_df[:split]
+df_val = metadata_df[split:]
+
+print(f"Size of the training set: {len(df_train)}")
+print(f"Size of the training set: {len(df_val)}")
+
+
+"""
+## Preprocessing
+"""
+
+"""
+Now, let's create a class DataGenerator, which will be used for
+real-time data feeding to the Keras model. We make the latter
+inherit the properties of `keras.utils.Sequence` class so that
+we can leverage nice functionalities.
+
+"""
+
+
+class DataGenerator(keras.utils.Sequence):
```

Comment:
Why did you switch away from using tf.data? tf.data should be preferred when possible. Sequence is more of a fallback.